### PR TITLE
global settings persistent + color by instance

### PR DESCRIPTION
**Pull request by lanzhenw**
## What changes are proposed in this pull request?
Screenshot 2023-09-22 at 10 58 00 AM

All settings on the global settings tab in color panel can persist in the same way as colorpool. 
User can save the settings in `dataset.app_config.colorscheme`.

Bugs: 
- [x] Opening the modal of quickstart-group fails. 

## How is this patch tested? If it is not, please explain why.
(Details)

### Is this a user-facing change that should be mentioned in the release notes?

-   [ ] No. You can skip the rest of this section.
-   [x] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [x] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [x] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

### CODE REVIEW

File: app/packages/core/src/components/ColorModal/ColorFooter.tsx [hunk 1]
```diff
@@ -31,7 +31,7 @@ const ColorFooter: React.FC = () => {
     useMutation<foq.setDatasetColorSchemeMutation>(foq.setDatasetColorScheme);
   const colorScheme = useRecoilValue(fos.colorScheme);
   const datasetName = useRecoilValue(fos.datasetName);
-  const defaultColorPool = useRecoilValue(fos.config).colorPool;
+  const configDeafult = useRecoilValue(fos.config);
```
Comment (benjaminpkane) [inline app/packages/core/src/components/ColorModal/ColorFooter.tsx:34]: small typo
```suggestion
  const configDefault = useRecoilValue(fos.config);
```

File: app/packages/core/src/components/ColorModal/ColorFooter.tsx [hunk 2]
```diff
@@ -89,7 +98,10 @@ const ColorFooter: React.FC = () => {
               setDatasetColorScheme({
                 variables: { subscription, datasetName, colorScheme: null },
               });
-              setColorScheme({ fields: [], colorPool: defaultColorPool });
+              setColorScheme({
+                fields: [],
+                colorPool: useRecoilValue(fos.config).colorPool,
```
Comment (benjaminpkane) [inline app/packages/core/src/components/ColorModal/ColorFooter.tsx:103]: This hook is conditionally run. I think the config is already available  via `configDefault`

File: app/packages/core/src/components/ColorModal/ColorModal.tsx [hunk 3]
```diff
@@ -22,6 +22,7 @@ import {
 import SidebarList from "./SidebarList";
 import { activeColorEntry } from "./state";
 import { ACTIVE_FIELD } from "./utils";
+import styles from "../../../../../packages/components/src/scrollable.module.css";
```
Comment (benjaminpkane) [inline app/packages/core/src/components/ColorModal/ColorModal.tsx:25]: Is it possible to use `import { scrollbarStyles } from "@fiftyone/components"`? The styles may need to be added as an export in  `@fiftyone/components/index.ts`

File: fiftyone/server/color.py [hunk 4]
```diff
@@ -88,6 +98,21 @@ def run():
             dataset.app_config.color_scheme = (
                 foo.ColorScheme(
                     color_pool=color_scheme.color_pool,
+                    color_by=color_scheme.color_by
```
Comment (benjaminpkane) [inline fiftyone/server/color.py:101]: Is there an issue with providing these values without a truthy check? 
```suggestion
                    color_by=color_scheme.color_by,
```

File: app/packages/core/src/components/ColorModal/controls/ColorAttribute.tsx [hunk 5]
```diff
@@ -58,24 +58,27 @@ const ColorAttribute: React.FC<Prop> = ({ style }) => {
...
   const activeField = useRecoilValue(activeColorField);
-  const { colorPool, fields } = useRecoilValue(fos.colorScheme);
-  const index = fields.findIndex((s) => s.path == activeField.path);
+  const currentColorScheme = useRecoilValue(fos.colorScheme);
+  const index = currentColorScheme.fields.findIndex(
+    (s) => s.path == activeField.path
+  );
 
   const options = subfields.map((field) => ({
     value: field.path?.split(".").slice(-1)[0],
```
Comment (benjaminpkane) [inline app/packages/core/src/components/ColorModal/controls/ColorAttribute.tsx:67]: nit: `field.path?.split(".").slice(-1)[0]` is used here and on line 72. Maybe the map function could hold it as a constant?

File: app/packages/core/src/components/ColorModal/colorPalette/AttributeColorSetting.tsx [hunk 6]
```diff
@@ -63,28 +63,34 @@ type Input = {
...
 const AttributeColorSetting: React.FC<ColorPickerRowProps> = ({
   style,
   useLabelColors,
 }) => {
   const pickerRef = useRef<ChromePicker>(null);
   const wrapperRef = useRef<HTMLDivElement>(null);
   const activeField = useRecoilValue(activeColorField);
-  const { colorPool = [], fields } = useRecoilValue(fos.colorScheme);
+  const colorScheme = useRecoilValue(fos.colorScheme);
+  // const { colorPool = [], fields } = useRecoilValue(fos.colorScheme);
```
Comment (benjaminpkane) [inline app/packages/core/src/components/ColorModal/colorPalette/AttributeColorSetting.tsx:76]:
```suggestion
```

**Review by benjaminpkane — COMMENTED** [review]
Awesome to see the color scheme settings growing! 🚀  Left some small comments

File: app/packages/core/src/components/ColorModal/ColorFooter.tsx (hunk 2 above)
Comment (lanzhenw) [reply app/packages/core/src/components/ColorModal/ColorFooter.tsx:103]: Good catch. Thank you!

File: app/packages/core/src/components/ColorModal/ColorModal.tsx (hunk 3 above)
Comment (lanzhenw) [reply app/packages/core/src/components/ColorModal/ColorModal.tsx:25]: Yes, it's already exported.

**Review by benjaminpkane — APPROVED** [review]
LGTM! 🖌️ e2e is more or less working now

**Pull request merged**